### ISSUE TITLE
Admin Password keeps resetting

### ISSUE THREAD

**Issue by ken8521** [filebrowser/filebrowser#1838]
I installed Filebrowser in a container, and it works as expected.  The problem comes when the container updates, restarts, or the compose file is ran again for updates.. my admin password resets.  This is an issue because I reverse proxy file browser through my domain, and if watchtower pulls an update and I don't know it or even the container restarts... someone who comes across it could easily log in with the default credentials.  I'm assuming I'm missing something here... but I can't find what.

  version: "2"
services:
  filebrowser:
    image: filebrowser/filebrowser:latest
    container_name: filebrowser  
    volumes:
      - /path/to:/data
      - /path/to/data:/srv
    ports:
      - 8080:80
    restart: unless-stopped 

Oddly enough, I just noticed the data folder is never written to.. it's always empty, so I think that might be part of the problem (maybe my paths are wrong.. but that was the instructions I found online).  For now I've taken it off line until I can figure this out.

Thanks for any help you can offer.

**Comment by ramiresviana** [comment]
Filebrowser does not make use of /data directory. When the container is destroyed, all files on it are deleted. User credentials are stored on the database file, you want to ensure that this file is persisted on container changes. For more information, check the documentation (https://filebrowser.org/) and Dockerfile (https://github.com/filebrowser/filebrowser/blob/master/Dockerfile).

**Comment by mptpro** [comment]
I have the same problem, whenver the Docker container is updated the Admin user resets, and other users are destroyed. Other settings, such as file paths, are retained.  I am not using a /data directory.

**Comment by DCMarine** [comment]
Same problem here, plus all created user accounts are removed, and any customizations made. 

after doing some digging I found the problem.    the database is being mapped incorrectly (because we followed the instructions).    The below is my docker run command, as you can see I had to map the database to a different location.

```
docker run -d --init \
    --restart=unless-stopped \
    -p 80:80 \
    -e TZ=America/New_York \
    -v /opt/filebroswer/files:/srv \
    -v /opt/filebroswer/database.db:/database.db \
    -v /opt/filebroswer/settings.json:/config/settings.json \
    -e PUID=$(id -u) \
    -e PGID=$(id -g) \
    --name filebrowser \
    filebrowser/filebrowser:latest
```

**Comment by jame25** [comment]
I also have this issue. Service works as expected, until I restart the container, then all settings are lost.

Here is my docker compose:
```
  filebrowser:
    image: filebrowser/filebrowser:latest
    network_mode: "service:gluetun"
    environment:
      - PUID=1000
      - PGID=1000
    volumes:
      - /media/hdd/Files:/srv
      - /home/plex/Docker/filebrwsr/database.db:/database/database.db
      - /home/plex/Docker/filebrwsr/.filebrowser.json:/.filebrowser.json
    restart: unless-stopped
```

**Comment by NolandTech** [comment]
> I also have this issue. Service works as expected, until I restart the container, then all settings are lost.
> 
> Here is my docker compose:
> 
> ```
>   filebrowser:
>     image: filebrowser/filebrowser:latest
>     network_mode: "service:gluetun"
>     environment:
>       - PUID=1000
>       - PGID=1000
>     volumes:
>       - /media/hdd/Files:/srv
>       - /home/plex/Docker/filebrwsr/database.db:/database/database.db
>       - /home/plex/Docker/filebrwsr/.filebrowser.json:/.filebrowser.json
>     restart: unless-stopped
> ```

Man that looks messed up, why do you have filebrowser inside your Docker directory, inside your Plex directory... It should be setup: 
home/USERNAME/docker/plex
home/USERNAME/docker/plex/sonarr
home/USERNAME/docker/plex/radarr

home/USERNAME/docker/filebrowser

**Comment by NolandTech** [comment]
I fixed it... 
  # filebrowser:
  #   image: filebrowser/filebrowser:latest
  #   container_name: filebrowser
  #   restart: always
  #   volumes:
  #     - ~/:/srv
  #     - ./filebrowser/filebrowser.db:/database/database.db
  #     - ./filebrowser/.filebrowser.json:/.filebrowser.json
  #     - ./filebrowser/branding:/branding
  #   environment:
  #     - PUID=1000
  #     - PGID=1000
  #     - FB_CONFIG="/.filebrowser.json"
  #     - FB_DATABASE="/database/filebrowser.db"
  #   ports:
  #     - "8085:8085"
  #   networks:
  #     - pi-hole_pi-hole
  
Use SFTP instead :)

**Comment by Guiorgy** [comment]
@jame25 as per the docs, the database file is located in the root directory, in other words `/database.db`, not `/database/database.db`. Alternatively, you can try using the `filebrowser/filebrowser:s6` image that suposedly has the database in `/database/filebrowser.db` and settings in `/config/settings.json`, though I caven't tried.

**Comment by Guiorgy** [comment]
If you don't want to use the `filebrowser/filebrowser:s6` image, and you don't want to bind those files individually, one possible solution is to override the entrypoint of the container, where you'd move the desired files into the volume and create symbolic links in their original places:

```yaml
volumes:
      - ...:/srv
      - filebrowser_data:/data
    entrypoint: >
      /bin/sh -c '
        if [ ! -e /data/database.db ]; then
          if [ ! -e /database.db ]; then
            /filebrowser &
            FB_PID=$!
            sleep 5 && kill -s SIGTERM $FB_PID
          fi &&
          mv /database.db /data/database.db
        fi &&
        if [ ! -L /database.db ]; then
          if [ -e /database.db ]; then
            rm /database.db
          fi &&
          ln -s /data/database.db /database.db
        fi &&
        if [ ! -e /data/.filebrowser.json ]; then
          mv /.filebrowser.json /data/.filebrowser.json
        fi &&
        if [ ! -L /.filebrowser.json ]; then
          if [ -e .filebrowser.json ]; then
            rm .filebrowser.json
          fi &&
          ln -s /data/.filebrowser.json /.filebrowser.json
        fi &&
        exec /filebrowser
      '
```

After running the above, I execute a shell inside the container and check the results:

```shell
/ # ls -la /database.db
lrwxrwxrwx         /database.db -> /data/database.db
/ # ls -la /.filebrowser.json 
lrwxrwxrwx         /.filebrowser.json -> /data/.filebrowser.json
/ # ls -la /data
total 76
drwxr-xr-x         .
drwxr-xr-x         ..
-rw-r--r--         .filebrowser.json
-rw-------         database.db
```

**Comment by ncodeempresarial** [comment]
Hello everyone,

I wanted to share a simple solution that worked for me — I was deploying FileBrowser using Coolify and encountered the same issue many of you reported.

Like @jame25  and @DCMarine, I noticed that the filebrowser.db file wasn't being detected, and the logs would say:

Warning: filebrowser.db can't be found. Initialing in /var/www/filebrowser/database.db

This happens because the database file (filebrowser.db) — which is a binary — doesn't exist either locally or in the GitHub repository when the container starts. When using docker-compose, if the file isn't present, FileBrowser just initializes a fresh one. But that also means you lose any saved users, settings, or data.

My fix:
I ran the Docker project locally, let FileBrowser generate the correct filebrowser.db, and then downloaded that file. I placed the file in the correct path locally and committed it to the repository.

On the next deploy with Coolify, everything worked as expected. Here's what the logs showed:

2025-04-26T03:27:15.914231986Z 2025/04/26 03:27:15 Using database: /database/filebrowser.db
2025-04-26T03:27:15.920374398Z 2025/04/26 03:27:15 Using config file: /.filebrowser.json
2025-04-26T03:27:15.941281690Z 2025/04/26 03:27:15 Listening on [::]:8081

So if you're seeing this issue, make sure to include a valid copy of filebrowser.db in your project — either by generating it locally or copying it from a previous instance — and mount it correctly in your docker-compose.yml.
